### PR TITLE
Safeguard against schedule entry without kwargs

### PR DESCRIPTION
**Pull request by auvipy**
Fixes ##5889

### CODE REVIEW

File: celery/beat.py [hunk 1]
```diff
@@ -381,7 +381,7 @@ def apply_async(self, entry, producer=None, advance=True, **kwargs):
 
         try:
             entry_args = [v() if isinstance(v, BeatLazyFunc) else v for v in (entry.args or [])]
-            entry_kwargs = {k: v() if isinstance(v, BeatLazyFunc) else v for k, v in entry.kwargs.items()}
+            entry_kwargs = {k: v() if isinstance(v, BeatLazyFunc) else v for k, v in (entry.kwargs.items() or {})}
```
Comment (thedrow) [inline celery/beat.py:384]: How can this be right?
`kwargs` is `None` so it has no `items()`.
You should check if `kwargs` is `None` and if it is call `items()` on an empty dictionary.

```suggestion
            entry_kwargs = {k: v() if isinstance(v, BeatLazyFunc) else v for k, v in (entry.kwargs or {}).items()}
```

**Review by thedrow — CHANGES_REQUESTED** [review]
This also needs tests.

File: celery/beat.py (hunk 1 above)
Comment (auvipy) [reply celery/beat.py:384]: I wasn't fully sure so wanted more eyes for some suggestion O:) thanks

File: celery/beat.py [hunk 2]
```diff
@@ -381,7 +381,7 @@ def apply_async(self, entry, producer=None, advance=True, **kwargs):
 
         try:
             entry_args = [v() if isinstance(v, BeatLazyFunc) else v for v in (entry.args or [])]
-            entry_kwargs = {k: v() if isinstance(v, BeatLazyFunc) else v for k, v in entry.kwargs.items()}
+            entry_kwargs = {k: v() if isinstance(v, BeatLazyFunc) else v for k, v in (entry.kwargs or {}).items()}
```
Comment (xirdneh) [inline celery/beat.py:384]: **[suggestion]**: Refactor this function for better readability. Avoid doing an extra boolean operation inside a comprehension. Avoid calling `items()` on an empty dict.

```python
def apply_async(self, entry, producer=None, advance=True, **kwargs):
    # Update time-stamps and run counts before we actually execute,
    # so we have that done if an exception is raised (doesn't schedule
    # forever.)
    entry = self.reserve(entry) if advance else entry
    task = self.app.tasks.get(entry.task)

    try:
        entry_args = self._evaluate_entry_args(entry.args)
        entry_kwargs = self._evaluate_entry_kwargs(entry.kwargs)

        self.send_task(entry, entry_args, entry_kwargs, task, producer)
    except Exception as exc:  # pylint: disable=broad-except
        reraise(SchedulingError, SchedulingError(
            "Couldn't apply scheduled task {0.name}: {exc}".format(
                entry, exc=exc)), sys.exc_info()[2])
    finally:
        self._tasks_since_sync += 1
        if self.should_sync():
            self._do_sync()

def _evaluate_entry_args(entry_args):
    if not entry_args:
        return []
    return [v() if isinstance(v, BeatLazyFunc) else v for v in entry_args]

def _evaluate_entry_kwargs(entry_kwargs):
    if not entry_kwargs:
        return {}
    return {k: v() if isinstance(v, BeatLazyFunc) else v for k, v in entry_kwargs}

# Repurpose existing function
def send_task(self, entry, entry_args, entry_kwargs, task, producer):
        if task:
            return task.apply_async(
                entry_args, entry_kwargs,
                producer=producer,
                **entry.options
            )
        else:
            return self.app.send_task(
                entry.task,
                entry_args,
                entry_kwargs,
                producer=producer,
                **entry.options
            )

**Review by xirdneh — CHANGES_REQUESTED** [review]
This also needs a test.

File: celery/beat.py (hunk 2 above)
Comment (thedrow) [reply celery/beat.py:384]: I agree.
@xirdneh Care to fix this?
@auvipy What about the test?

**Review by thedrow — APPROVED** [review]

**Comment by thedrow** [comment]
Codecov is wrong and is reporting on unrelated files.

**Comment by xirdneh** [comment]
If this only happens on `advance=False` then it looks like the problem is somewhere before this code.
This might not fix the root cause but it certainly fixes this specific problem.
The issue is that this  problem where `entry.kwargs == None` might show up somewhere else.

Should we create an issue to make sure that `entry` has correct values?

**Comment by thedrow** [comment]
> Should we create an issue to make sure that `entry` has correct values?

Yes we should.

**Review by thedrow — APPROVED** [review]

**Pull request merged**